doc/gambit.txi

```diff
@@ -13426,7 +13423,26 @@ $ @b{./foo 1 2 "3 4"}
 
 @end deffn
 
-@node Environment variables, Measuring time, Command line arguments, Host environment
+@deffn procedure executable-path
+
+This procedure returns a newly allocated string representing a
+@dfn{normalized path} to the executable file from which the running process
+spawned.  The path is determined dynamically using a mechanism specific
+to the host operating system.
+
+For example under UNIX:
+
+@smallexample
+$ @b{echo '(pp (executable-path))' > ep.scm}
+$ @b{gsc -exe ep.scm}
+$ @b{mv ep $(mktemp -d)}
```

Comment:
If possible, examples should be executable so that they can be used as unit tests. So it would be good to write this example so that its execution is deterministic. Unfortunately using `mktemp`, and `/tmp` in general, is probably not deterministic. One way would be to reveal the CWD with a `pwd`, and the unit testing framework could enforce that the CWD for examples is always the same, say `/gambit/example`. This would give:

```
$ pwd
/gambit/example
$ ./ep
/gambit/example/ep
```

Reply:
Makes sense, thank you for clearing that up. Had gone out of my way there to _not_ use a determinisitc example.